wlct/management/commands/engine.py

```diff
@@ -72,6 +74,94 @@ def parse_and_update_clan_logo():
+                return True
+        return False
+    except:
+        log_exception()
+        return False
+
+
+def is_valid_team_order(team_order, team_dict, teams):
+    for i in range(len(team_order)):
+        for player in team_dict[team_order[i]]:
+            if is_correct_player(player, teams[i]):
+                return True
+    return False
+
+def validate_cl_tournament_games():
+    try:
+        print("Running validate_cl_tournament_games on thread {}".format(threading.get_ident()))
+        tournament_games = TournamentGame.objects.filter(is_finished=False)
+        api = API()
+        found_missing_games = 0
+        corrected_missing_games = 0
+        test_content = TestContent()
+
+        # Iterate through unfinished games and check of missing player values
+        for game in tournament_games:
+            teams = game.teams.split('.')
+            tt = TournamentTeam.objects.filter(pk=teams[0])
+            if not tt or tt[0].clan_league_clan:
+                continue
+            if not game.players:
+                log("Found TournamentGame missing 'players' value with id: {}".format(game.id), LogLevel.engine)
+                found_missing_games += 1
+                test_msg = test_content.team_game(teams[0], teams[1])
+                api_response = api.api_query_game_feed(game.gameid, test_msg).json()
+
+                # Add player tokens by team
+                team_dict = {}
+                team_order = []
+                for player in api_response.get("players"):
+                    if "team" in player:
+                        if not player["team"] in team_dict:
+                            team_dict[player["team"]] = []
+                            team_order.append(player["team"])
+                        team_dict[player["team"]].append(player["id"])
+                    else:
+                        if not len(team_dict) in team_dict:
+                            team_dict[len(team_dict)] = []
+                            team_order.append(len(team_dict))
+                        team_dict[len(team_dict) - 1].append(player["id"])
+                # Check if team order matches with player and clan object on clot
+                if not is_valid_team_order(team_order, team_dict, teams):
+                    log("Possibly incorrect team order... Attempting reversing teams - TGame ID: {}".format(game.id), LogLevel.warning)
+                    team_order.reverse()
+                    if not is_valid_team_order(team_order, team_dict, teams):
+                        log("Possibly incorrect team order... Using original order - TGame ID: {}".format(game.id), LogLevel.warning)
+                        team_order.reverse()
+                    else:
+                        log("Correct team order found - TGame ID: {}".format(game.id), LogLevel.engine)
+                else:
+                    log("Correct team order found - TGame ID: {}".format(game.id), LogLevel.engine)
```

Comment:
Not something to fix here, but we need a better way to search the logs. I think I need to expand on the logger class to take in more than keywords/categories, to enable better filtering as wlel.